src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -345,11 +405,32 @@ private void UpdateCurrentConfiguration()
             _onBackgroundTaskFinish?.Invoke();
         }
 
-        private void UpdateConfiguration(T configuration)
+        private void UpdateConfiguration(T configuration, DateTimeOffset retrievalTime)
         {
             _currentConfiguration = configuration;
-            _syncAfter = DateTimeUtil.Add(TimeProvider.GetUtcNow().UtcDateTime, AutomaticRefreshInterval +
+            _syncAfter = DateTimeUtil.Add(retrievalTime.UtcDateTime, AutomaticRefreshInterval +
                 TimeSpan.FromSeconds(new Random().Next((int)AutomaticRefreshInterval.TotalSeconds / 20)));
+
+            if (ConfigurationEventHandler != null)
+            {
+                _ = Task.Run(async () =>
+                {
+                    try
+                    {
+                        await ConfigurationEventHandler.AfterUpdateAsync(MetadataAddress, configuration).ConfigureAwait(false);
+                    }
+                    catch (Exception ex)
+                    {
+                        LogHelper.LogExceptionMessage(
+                            new InvalidOperationException(
+                                LogHelper.FormatInvariant(
+                                   LogMessages.IDX20812,
```

Comment:
Wrong error message?

`internal const string IDX20812 = "IDX20812: Failed to validate configuration retrieved via event handler. Exception: '{0}'.";`

Reply:
Good catch, fixing this.